From 5afc5554f30a8f7bc2ce319fe1aed43e7769b358 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Klime=C5=A1?= <jklimes@redhat.com>
Date: Wed, 24 Nov 2010 17:24:28 +0100
Subject: [PATCH] ifcfg-rh: allow missing and 0.0.0.0 GATEWAYn lines in
 ifcfg-routes (rh #647992)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GATEWAY0=0.0.0.0 was erroneously denied. Also, missing GATEWAY0 entry caused
ifcfg-rh plugin to regard the connection as invalid. The commit fixes that and
makes it behave in accordance with initscripts.

Signed-off-by: Jiří Klimeš <jklimes@redhat.com>

(cherry picked from commit 5afc5554f30a8f7bc2ce319fe1aed43e7769b358)
---
 system-settings/plugins/ifcfg-rh/reader.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/system-settings/plugins/ifcfg-rh/reader.c b/system-settings/plugins/ifcfg-rh/reader.c
index 8c4825b..bca1818 100644
--- a/system-settings/plugins/ifcfg-rh/reader.c
+++ b/system-settings/plugins/ifcfg-rh/reader.c
@@ -722,12 +722,7 @@ read_one_ip4_route (shvarFile *ifcfg,
 	/* Next hop */
 	if (!read_ip4_address (ifcfg, gw_tag, &tmp, error))
 		goto out;
-	if (!tmp) {
-		g_set_error (error, IFCFG_PLUGIN_ERROR, 0,
-		             "Missing or invalid IP4 gateway address '%d'",
-		             tmp);
-		goto out;
-	}
+	/* No need to check tmp, because we don't make distinction between missing GATEWAY IP and 0.0.0.0 */
 	nm_ip4_route_set_next_hop (route, tmp);
 
 	/* Prefix */
-- 
1.7.11.7

